src/main/java/com/odysseusinc/athena/service/mail/EmailSenderService.java

```diff
@@ -35,50 +36,62 @@
 import javax.mail.MessagingException;
 import javax.mail.internet.MimeMessage;
 import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
 
+import static com.google.common.collect.Iterables.toArray;
+import static java.util.Collections.emptyList;
+
 @Service
 public class EmailSenderService {
-    private static final Logger LOGGER = LoggerFactory.getLogger(EmailSenderService.class);
+    private static final Logger log = LoggerFactory.getLogger(EmailSenderService.class);
 
-    private final MailContentBuilder contentBuilder;
     private final JavaMailSender mailSender;
-
-    @Value("${athena.mail.notifier}")
-    private String notifier;
-
-    @Value("${spring.mail.username}")
-    private String from;
+    private final MailContentBuilder contentBuilder;
+    private final String from;
+    private final String notifier;
 
     @Autowired
-    public EmailSenderService(MailContentBuilder contentBuilder, JavaMailSender mailSender) {
+    public EmailSenderService(MailContentBuilder contentBuilder,
+                              JavaMailSender mailSender,
+                              @Value("${athena.mail.notifier}") String notifier,
+                              @Value("${spring.mail.username}") String from) {
 
         this.contentBuilder = contentBuilder;
         this.mailSender = mailSender;
+        this.notifier = notifier;
+        this.from = from;
     }
 
-    public void send(EmailType messageType, Map<String, Object> parameters, String... emails) {
+    public void send(EmailType messageType, Map<String, Object> parameters, String... toEmails) {
 
-        if (emails == null || emails.length == 0 || StringUtils.isBlank(emails[0])) {
+        this.send(messageType, parameters, Arrays.asList(toEmails), emptyList());
+    }
+
+
+    public void send(EmailType messageType, Map<String, Object> parameters, List<String> toEmails, List<String> bccEmails) {
+
+        if (CollectionUtils.isEmpty(toEmails) && CollectionUtils.isEmpty(bccEmails)) {
```

Comment:
I can see from the code that `toEmails` can be empty if `bccEmails` has values. Is this right? bcc without main-emails looks strange.

Reply:
I do not see a problem here